### PR TITLE
stage2: implement ** and ++ at runtime

### PR DESCRIPTION
**Pull request by g-w1**

### CODE REVIEW

File: src/Sema.zig [hunk 1]
```diff
@@ -6526,12 +6528,70 @@ fn zirArrayCat(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Ai
...
+        try Type.Tag.array.create(sema.arena, .{ .len = final_len, .elem_type = lhs_info.elem_type });
+    const ptr_ty = try Type.ptr(sema.arena, .{
+        .pointee_type = array_ty,
+        .@"addrspace" = target_util.defaultAddressSpace(sema.mod.getTarget(), .local),
+    });
+    const alloc = try block.addTy(.alloc, ptr_ty);
+
+    // TODO there is some ownership issue where if one of these is comptime-known, it generates an invalid tv
+    const ptr_to_lhs = if (!is_pointer) try sema.analyzeRef(block, lhs_src, lhs) else lhs;
+    const ptr_to_rhs = if (!is_pointer) try sema.analyzeRef(block, rhs_src, rhs) else rhs;
```
Comment (g-w1) [inline src/Sema.zig:6548]: Do I have to copy one of these into sema.arena if it is comptime known?
I get invalid int values inside the air array.
image

Comment (g-w1) [reply src/Sema.zig]: I put in some code to handle it, and its still there ...

**Review by andrewrk — CHANGES_REQUESTED** [review]
In your crash screenshot, the Value object that has a bad memory reference is coming from a Decl's Value. Check the Decl name in a debugger and figure out why that Decl's Value is invalid.

**Comment by g-w1** [comment]
I stubbed the case that causes the uaf out as a TODO for now as it is confusing me.

**Comment by g-w1** [comment]
Ok, marking as draft as memset with bitcast is being weird in llvm backend.

**Comment by andrewrk** [comment]
30+ days old draft; closing.

**Pull request closed without merging**